### DISCUSSION TITLE
Reload too slow

### DISCUSSION THREAD

**Discussion by chentiangang** [prometheus/prometheus#12087] [Q&A]
### What did you do?

curl -X POST http://localhost:9090/-/reload

### What did you expect to see?
What I'm looking forward to is a quick and normal execution

### What did you see instead? Under which circumstances?

It takes me 15s-30s each time
I found this problem on CenOS 7.6.1810
When I went back to prometheus 2.8.1，There is no problem
I have found no such problems on mac OS with different versions.

### System information

Linux 3.10.0.957.e17.x86_64 x86_64

### Prometheus version

```text
prometheus, version 2.37.0 (branch: HEAD, revision: b41e0750abf5cc18d8233161560731de05199330)
  build user:       root@04e08efba15f
  build date:       20220714-15:13:17
  go version:       go1.18.4
  platform:         darwin/amd64
```

**Comment by kaffarell** [comment]
Could you please translate your issue to english, so that I can have a look at it...

**Comment by chentiangang** [comment]
thank you，This question has puzzled me for a long time

**Comment by kaffarell** [comment]
Could you post your config file here (Maybe block out secrets and stuff)? You can also try the newest version (2.39.1) maybe that will fix this.

**Comment by roidelapluie** [comment]
Can you enable debug logs and give us the config reload times for each section?

**Comment by chentiangang** [comment]
In the end, I didn't run into this problem on other linux versions.
